### PR TITLE
CFD-294 : Homepage : My Groups (Logged-in users).

### PR DESCRIPTION
**Pull request by nader77**
#478

### CODE REVIEW

**Comment by nader77** [comment]
Update:
Added all the functionality for my groups block.
Displays 3 groups in alphabetical order, The rest can be view when clicking on "Show all" button.

Custom view for this block displays:
* group-type icon
* group name (clickable, link to group dashboard)
* group owner (clickable, link to public person profile)
* number of group members

As specified in the Wireframe. @HelenaEksler You should use the same node view for "Suggested groups" in PR #480 (Both blocks should have the same view).

**Comment by nader77** [comment]
Block collapsed: 


Block expanded:
selection_047

There's no content type icon for groups in the theme.

File: capacity4more/modules/c4m/ds/c4m_ds/includes/c4m_ds.theme.node.inc [hunk 1]
```diff
@@ -276,6 +276,30 @@ function theme_c4m_ds_group_access_info($field) {
...
+function theme_c4m_ds_group_members_count($field) {
+  if (!isset($field['entity'])) {
+    return NULL;
+  }
+
+  $group_members = c4m_og_get_members_count($field['entity']);
+
+  $tag['element'] = array(
+    '#tag'   => 'span',
+    '#value' => $group_members . '&nbsp;' . t('members'),
```
Comment (amitaibu) [inline capacity4more/modules/c4m/ds/c4m_ds/includes/c4m_ds.theme.node.inc]: move everything inside the t() --  t('@count members')

File: capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ * Render callback.
+ */
+function c4m_features_homepage_my_groups_content_type_render($subtype, $conf, $args, $context) {
+  global $user;
+  $my_groups = array();
+  $first_groups = '';
+  $extra_groups = '';
+
+  // Get All user's groups (We get them sorted by id).
+  $groups = og_get_groups_by_user($user);
```
Comment (amitaibu) [inline capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc]: I think you can just ``og_get_groups_by_user()``

**Comment by nader77** [comment]
Update:
* Fixed syntax comments.
* Fixed the homepage layout.

My groups and homepage layout:
groups-2015-02-16_11

**Comment by nader77** [comment]
@HelenaEksler PR is ready for review.

**Comment by HelenaEksler** [comment]
@nader77  Looks fine. As soon as tests are green - assign to Peter.

File: capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  // Get All user's groups (We get them sorted by id).
+  $groups = og_get_groups_by_user();
+
+  if (empty($groups['node'])) {
+    return FALSE;
+  }
+
+  // Load node for each group and add the first 2 letters of the group title as a key.
+  foreach ($groups['node'] as $group) {
```
Comment (amitaibu) [inline capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc]: better do a ``node_load_multiple($groups['node'])``

Comment (amitaibu) [inline capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc]: better do a ``node_load_multiple($groups['node'])``

**Comment by nader77** [comment]
@amitaibu @HelenaEksler Fixed.

**Comment by zero2one** [comment]
@nader77 
When you login as the site administrator (admin): 

There is a difference between the number of groups on the profile page (8):
c4m-profile-groups

And the number of groups on the site homepage (6):
c4m-homepage-groups

**Comment by nader77** [comment]
@zero2one The other 2 are Projects, They should be displayed here ?

**Comment by zero2one** [comment]
@nader77 Ouch, sorry, did not notice it…

You are correct, there should only be groups in the list.

Can you do a re-roll?

**Comment by nader77** [comment]
@zero2one Re-rolled.

**Comment by zero2one** [comment]
Seems that every pulled-in pull request blocks the other pull requests about the home page…

@nader77  Can you re-roll please?

**Comment by HelenaEksler** [comment]
@zero2one Re-roled

**Comment by zero2one** [comment]
Needs a follow up for theming.

**Pull request merged**